analyses/infercnv-consensus-cell-type/template-notebooks/ewings-infercnv-results.Rmd

```diff
@@ -0,0 +1,536 @@
+    readr::read_tsv(ewings_tsv) |>
+      dplyr::select(barcodes, consensus_annotation, ewing_annotation)
+  )
+```
+
+
+## InferCNV Heatmaps
+
+Below, we show the `inferCNV` heatmaps from the `immune-all` and `immune-subset` normal references for this library.
+
+To help contextualize these, here is the table of known CNVs in Ewing sarcoma we'll use for validation:
+
+```{r}
+validation_df
+```
+
+### Reference with all immune cells
+
+![InferCNV heatmap with immune-all reference](`r immune_png`)
+
+### Reference with a subset of immune cells 
+
+![InferCNV heatmap with immune-subset reference](`r subset_png`)
+
+
+
+## Distribution of CNVs per cell
+
+This section contains figures showing the total number of CNVs detected per cell. 
+
+### CNVs in reference vs query cells
+
+The figures below compare the number of CNVs per cell between reference and query cells. 
+Ideally, we'll see a bimodal distribution where reference cells are at/near zero, and there is a peak of higher values which may represent tumor cells.
+In the density plots below, distributions are colored based on whether the cells are in the reference or not (aka, query cells).
+
+```{r}
+# First, construct data frame of total `has_cnv` counts, with other variables needed for plots
+has_cnv_df <- infercnv_df |>
+  dplyr::select(
+    barcodes,
+    normal_reference,
+    starts_with("has_cnv_")
+  ) |>
+  tidyr::pivot_longer(
+    starts_with("has_cnv_"),
+    names_to = "chr",
+    values_to = "cnv"
+  ) |>
+  # sum all values for each cell, but separately for each run of inferCNV (aka each normal reference)
+  dplyr::group_by(barcodes, normal_reference) |>
+  dplyr::summarize(total_cnv_per_cell = sum(cnv)) |>
+  dplyr::ungroup() |>
+  # add indicator columns for whether the cell is reference or query
+  dplyr::mutate(
+    immune_reference_cell = barcodes %in% immune_ref_barcodes,
+    subset_reference_cell = barcodes %in% subset_ref_barcodes
+  ) |>
+  # join in SCE information
+  dplyr::left_join(umap_df) |>
+  # finally, bring back in cluster and library information
+  dplyr::left_join(
+    infercnv_df |>
+      dplyr::select(barcodes, library_id, subcluster, normal_reference) |>
+      unique(),
+    by = c("barcodes", "normal_reference")
+  )
+```
+
+
+```{r fig.width = 10, fig.height = 5}
+all_plot <- has_cnv_df |>
+  dplyr::filter(normal_reference == "all-immune") |>
+  ggplot() +
+  aes(x = total_cnv_per_cell, fill = immune_reference_cell) +
+  geom_density(alpha = 0.5) +
+  labs(
+    title = "Immune-all reference"
+  ) +
+  theme(legend.position = "bottom")
+
+subset_plot <- has_cnv_df |>
+  dplyr::filter(normal_reference == "subset-immune") |>
+  ggplot() +
+  aes(x = total_cnv_per_cell, fill = subset_reference_cell) +
+  geom_density(alpha = 0.5) +
+  labs(
+    title = "Immune-subset reference"
+  ) +
+  theme(legend.position = "bottom")
+
+wrap_plots(all_plot, subset_plot) +
+  plot_annotation(title = "Total CNVs per cell")
+```
+
+### CNVs across cell types
```

Comment:
For this plot I would lump tumor cells together. Alternatively, maybe we should make a category that's either reference, non-reference normal, and tumor and use that to facet these plots.